src/libraries/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -16,6 +16,24 @@ internal static class IPAddressParser
         internal const int MaxIPv4StringLength = 15; // 4 numbers separated by 3 periods, with up to 3 digits per number
         internal const int MaxIPv6StringLength = 65;
 
+        public static unsafe bool IsValid<TChar>(ReadOnlySpan<TChar> ipSpan)
+            where TChar : unmanaged, IBinaryInteger<TChar>
+        {
+            fixed (TChar* ipStringPtr = &MemoryMarshal.GetReference(ipSpan))
+            {
+                if (ipSpan.Contains(TChar.CreateTruncating(':')))
+                {
+                    return IPv6AddressHelper.IsValidStrict(ipStringPtr, 0, ipSpan.Length);
+                }
+                else
+                {
+                    int end = ipSpan.Length;
+                    long address = IPv4AddressHelper.ParseNonCanonical(ipStringPtr, 0, ref end, notImplicitFile: true);
+                    return address != IPv4AddressHelper.Invalid && end == ipSpan.Length;
```

Comment:
What address does IPv4AddressHelper.Invalid represent?

Reply:
It's a -1 sentinel value